docker/docker_tools.py

```diff
@@ -800,9 +784,11 @@ def _build_phase2(arm: bool, dev: bool, pic24: bool, tex: bool, rust: bool):
 
     # Certbot requires nginx to be running to succeed, hence its placement here.
     if env.CERTBOT_EMAIL:
-        xqt(
-            'certbot -n --agree-tos --email "$CERTBOT_EMAIL" --nginx --redirect -d "$RUNESTONE_HOST"'
-        )
+        # See if there's already a certificate for this host. If not, get one.
+        if not Path(f"/etc/letsencrypt/live/{env.RUNESTONE_HOST}").is_dir():
```

Comment:
We **may** want to think about running `certbot renew` if the cert exists to avoid any staleness due to this being in a volume.

Reply:
Per the [docs](https://certbot.eff.org/docs/using.html#renewing-certificates), "renew is suitable (and designed) for automated use, to allow your system to automatically renew each certificate when appropriate. Since renew only renews certificates that are near expiry it can be run as frequently as you want - since it will usually take no action."

I'll go ahead and add this -- good idea.